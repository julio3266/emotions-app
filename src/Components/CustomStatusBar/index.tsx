import React from 'react';
import {StatusBar} from 'react-native';
import {assignTestId} from '@utils/qualityAssurance';

type TypeOfBar = 'light' | 'dark';

export interface ICustomStatusBar {
  testID?: string;
  type: TypeOfBar;
}

export const CustomStatusBar: React.FC<ICustomStatusBar> = ({
  testID = 'CustomStatusBar',
  type,
}) => {
  return (
    <>
      <StatusBar
        {...assignTestId('TouchableOpacity', `${testID}_backButton`)}
        barStyle={type === 'light' ? 'light-content' : 'default'}
      />
    </>
  );
};
