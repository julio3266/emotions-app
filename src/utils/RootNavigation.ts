import {createRef} from 'react';

export const navigationRef: any = createRef();

export const navigate = (name: string, params?: any) => {
  navigationRef.current?.navigate(name, params);
};

export const goBack = () => {
  navigationRef.current?.goBack();
};

export const params = () => {
  navigationRef.current?.params;
};
