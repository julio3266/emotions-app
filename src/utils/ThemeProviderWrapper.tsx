import React from 'react';
import {ThemeProvider} from 'styled-components/native';
import Theme from '../theme/styles';

interface IProps {
  children: any;
  padding?: boolean;
}

export const ThemeProviderWrapper: React.FC<IProps> = ({children}) => {
  return <ThemeProvider theme={Theme}>{children}</ThemeProvider>;
};
