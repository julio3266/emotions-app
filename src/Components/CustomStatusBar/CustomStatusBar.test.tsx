import React from 'react';
import {render} from '@testing-library/react-native';
import {mockTestID} from '../../utils/qualityAssurance';
import {ThemeProviderWrapper as wrapper} from '../../utils/ThemeProviderWrapper';
import {CustomStatusBar, ICustomStatusBar} from '@Components/CustomStatusBar';

describe('Components/CustomStatusBar', () => {
  let props: ICustomStatusBar = {
    testID: 'CustomStatusBar',
    type: 'light',
  };
  const {findByTestId, toJSON} = render(<CustomStatusBar {...props} />, {
    wrapper,
  });

  const sut_container = mockTestID('View', props.testID!).testID;

  it(`should render #${sut_container}`, () => {
    const sut = findByTestId(sut_container);
    expect(sut).toBeTruthy();
  });

  it(`should render a snapshot`, () => {
    expect(toJSON()).toMatchSnapshot();
  });
});
