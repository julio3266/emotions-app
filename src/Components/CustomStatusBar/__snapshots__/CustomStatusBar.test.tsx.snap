// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/CustomStatusBar should render a snapshot 1`] = `null`;
